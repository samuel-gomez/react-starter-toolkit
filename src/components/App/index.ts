export { default } from './App.container';
export type { TApp } from './App';
