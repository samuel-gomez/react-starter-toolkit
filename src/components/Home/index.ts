export { default } from './Home';
export type { THome } from './Home';
