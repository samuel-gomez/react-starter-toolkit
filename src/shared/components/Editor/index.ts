export { withEditor, useEditable } from './Editor';
export type { TEvent } from './Editor';
