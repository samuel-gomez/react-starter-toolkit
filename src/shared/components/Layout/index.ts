export { default } from './Layout';
export type { TLayout, TLayoutPage } from './Layout';
