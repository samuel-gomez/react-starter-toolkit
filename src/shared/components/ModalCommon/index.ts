export { default as ModalCommonBody } from './ModalCommonBody';
export type { TModalCommonBody } from './ModalCommonBody';
export { default as ModalCommonHeader } from './ModalCommonHeader';
export type { TModalCommonHeader } from './ModalCommonHeader';
export { default as ModalCommonFooter } from './ModalCommonFooter';
export type { TModalCommonFooter } from './ModalCommonFooter';
export { default as useToggleModal } from './ModalCommon.hook';
export type { TReturnUseToggleModal } from './ModalCommon.hook';
