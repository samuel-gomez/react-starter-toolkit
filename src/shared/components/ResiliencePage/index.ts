export { default } from './ResiliencePage.container';
export type { TResiliencePageContainer } from './ResiliencePage.container';
