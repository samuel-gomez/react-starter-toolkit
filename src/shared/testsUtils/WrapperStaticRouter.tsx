/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
