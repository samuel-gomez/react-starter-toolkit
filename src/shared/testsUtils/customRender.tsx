/* eslint-disable react/display-name */
import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import FetchProvider from 'providers/FetchProvider';
import UserProvider from 'providers/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <MemoryRouterProvider url="">{children}</MemoryRouterProvider>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render };
