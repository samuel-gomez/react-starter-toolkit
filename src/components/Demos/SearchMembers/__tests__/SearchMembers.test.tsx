import { screen } from '@testing-library/react';
import { MODES } from 'shared/components/Loader';
import { emptyFunction, renderWithWrapperStaticRouter, WrapperQuery } from 'shared/testsUtils';
import SearchMembers, { getDownloadPath, setFileName, DownloadLinkEnhanced } from '../SearchMembers';

describe('getDownloadPath', () => {
  it('Should return "members/0064962455/download-detail" When getDownloadPath with memberId "0064962455"', () => {
    const result = getDownloadPath('0064962455');
    const expected = 'members/0064962455/download-detail';
    expect(result).toEqual(expected);
  });
});

describe('setFileName', () => {
  it('Should return prefix___20180922.csv When setFileName with date (2018, 8, 22)', () => {
    const result = setFileName({ date: `${new Date(2018, 8, 22)}` });
    const expected = 'prefix___20180922.csv';
    expect(result).toEqual(expected);
  });

  it('Should return prefix___20180922.csv When setFileName with date (2018, 8, 22)', () => {
    const result = setFileName({ name: 'name', memberId: '0123456789', date: `${new Date(2018, 8, 22)}` });
    const expected = 'prefix_name_0123456789_20180922.csv';
    expect(result).toEqual(expected);
  });

  it('Should return prefix___20180922.csv When setFileName with formatDateFn return 20180922', () => {
    const formatDateFnMock = jest.fn().mockReturnValue('20180922');
    const result = setFileName({ name: 'name', memberId: '0123456789', formatDateFn: formatDateFnMock });
    const expected = 'prefix_name_0123456789_20180922.csv';
    expect(result).toEqual(expected);
  });
});

describe('<DownloadLinkEnhanced />', () => {
  const setFileNameFnMock = jest.fn();
  const getDownloadPathFnMock = jest.fn();
  const defaultDownloadProps = {
    idKey: '09878817',
    firstname: 'samuel',
    lastname: 'gomez',
  };
  it('Should call getDownloadPathFn and setFileNameFn with proper params', () => {
    const { asFragment } = renderWithWrapperStaticRouter(
      <WrapperQuery>
        <DownloadLinkEnhanced {...defaultDownloadProps} getDownloadPathFn={getDownloadPathFnMock} setFileNameFn={setFileNameFnMock} />
      </WrapperQuery>,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getDownloadPathFnMock).toBeCalledWith('09878817');
    expect(setFileNameFnMock).toBeCalledWith({ memberId: '09878817', name: 'samuel-gomez' });
  });

  it('Should call getDownloadPathFn and setFileNameFn with proper params 2 ', () => {
    const { asFragment } = renderWithWrapperStaticRouter(
      <WrapperQuery>
        <DownloadLinkEnhanced {...defaultDownloadProps} />
      </WrapperQuery>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

const defaultProps = {
  members: [],
  anomaly: null,
  loaderMode: MODES.none,
  submitFormSearchMembers: emptyFunction,
};

describe('<SearchMembers/>', () => {
  it('Should render <SearchMembers /> with one member', () => {
    renderWithWrapperStaticRouter(
      <SearchMembers
        {...defaultProps}
        searchMembers={[
          {
            key: '1',
            cols: {
              firstname: {
                label: 'nom du membre',
              },
              lastname: {
                label: 'nom du membre',
              },
              _id: {
                label: '1254689759',
              },
              actions: {
                children: <p>hello</p>,
                classModifier: 'actions',
              },
            },
          },
        ]}
      />,
    );
    const hello = screen.getByText('hello');
    expect(hello).toBeInTheDocument();
    expect(hello.closest('td')).toHaveClass('af-table__cell--actions');
  });
});
