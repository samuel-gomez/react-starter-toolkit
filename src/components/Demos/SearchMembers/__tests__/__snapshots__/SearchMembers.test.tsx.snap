// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownloadLinkEnhanced /> Should call getDownloadPathFn and setFileNameFn with proper params 1`] = `
<DocumentFragment>
  <div
    class="af-loader af-loader--fullscreen"
  >
    <button
      aria-label="Télécharger"
      class="af-link af-link--hasIconLeft af-link--download"
      type="button"
    >
      <svg
        aria-labelledby="title"
        class="af-download-link__icon"
        role="img"
        viewBox="0 0 100 100"
      >
        <title>
          download-csv
        </title>
        <path
          d="M70.4 5H13.9v90h72.2V20.7L70.4 5zM83 91.2H50v-7.8h24.7V67.8H49.9v-5.2L63 39.8h-6.8V16.9H43.5v22.9h-6.8l13.1 22.7v5.2H25v15.6h24.8v7.8H16.7V8.4h53.8v12.3H83v70.5zM49.7 76.3l-1-.4c-.8-.3-1.6-1-1.6-2.2 0-1.3 1.2-2.4 2.9-2.4.9 0 1.9.3 2.6 1l-1 1.3c-.5-.4-1-.6-1.6-.6-.5 0-.8.2-.8.5 0 .4.4.5 1.2.8l.9.4c1 .4 1.5 1.1 1.5 2.1 0 1.3-1.1 2.5-3 2.5-1 0-2.1-.4-3-1.1l1.2-1.4c.6.4 1.2.8 1.8.8.6 0 .9-.2.9-.6 0-.3-.4-.4-1-.7zm3.4-4.8h2.2l.8 3.4c.2.8.4 1.6.6 2.5.2-.8.4-1.6.6-2.5l.8-3.4h2.1l-2.3 7.8h-2.5l-2.3-7.8zm-9 6.1c.6 0 1.1-.3 1.4-.7l1.1 1.3c-.7.8-1.6 1.2-2.6 1.2-2 0-3.7-1.3-3.7-4 0-2.6 1.8-4.1 3.8-4.1 1 0 1.9.5 2.4 1l-1.1 1.3c-.4-.3-.8-.6-1.3-.6-.9 0-1.7.8-1.7 2.3 0 1.5.7 2.3 1.7 2.3z"
        />
      </svg>
      <span
        class="af-link__text"
      >
        Télécharger
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`<DownloadLinkEnhanced /> Should call getDownloadPathFn and setFileNameFn with proper params 2  1`] = `
<DocumentFragment>
  <div
    class="af-loader af-loader--fullscreen"
  >
    <button
      aria-label="Télécharger"
      class="af-link af-link--hasIconLeft af-link--download"
      type="button"
    >
      <svg
        aria-labelledby="title"
        class="af-download-link__icon"
        role="img"
        viewBox="0 0 100 100"
      >
        <title>
          download-csv
        </title>
        <path
          d="M70.4 5H13.9v90h72.2V20.7L70.4 5zM83 91.2H50v-7.8h24.7V67.8H49.9v-5.2L63 39.8h-6.8V16.9H43.5v22.9h-6.8l13.1 22.7v5.2H25v15.6h24.8v7.8H16.7V8.4h53.8v12.3H83v70.5zM49.7 76.3l-1-.4c-.8-.3-1.6-1-1.6-2.2 0-1.3 1.2-2.4 2.9-2.4.9 0 1.9.3 2.6 1l-1 1.3c-.5-.4-1-.6-1.6-.6-.5 0-.8.2-.8.5 0 .4.4.5 1.2.8l.9.4c1 .4 1.5 1.1 1.5 2.1 0 1.3-1.1 2.5-3 2.5-1 0-2.1-.4-3-1.1l1.2-1.4c.6.4 1.2.8 1.8.8.6 0 .9-.2.9-.6 0-.3-.4-.4-1-.7zm3.4-4.8h2.2l.8 3.4c.2.8.4 1.6.6 2.5.2-.8.4-1.6.6-2.5l.8-3.4h2.1l-2.3 7.8h-2.5l-2.3-7.8zm-9 6.1c.6 0 1.1-.3 1.4-.7l1.1 1.3c-.7.8-1.6 1.2-2.6 1.2-2 0-3.7-1.3-3.7-4 0-2.6 1.8-4.1 3.8-4.1 1 0 1.9.5 2.4 1l-1.1 1.3c-.4-.3-.8-.6-1.3-.6-.9 0-1.7.8-1.7 2.3 0 1.5.7 2.3 1.7 2.3z"
        />
      </svg>
      <span
        class="af-link__text"
      >
        Télécharger
      </span>
    </button>
  </div>
</DocumentFragment>
`;
